src/_common/content/components/media-item/media-item.ts

```diff
@@ -13,15 +13,19 @@ import AppContentViewerTS from '../../content-viewer/content-viewer';
 import AppContentViewer from '../../content-viewer/content-viewer.vue';
 import AppBaseContentComponent from '../base/base-content-component.vue';
 
-export function createMediaItemFromContentOwner(owner: ContentOwner, mediaItemId: number) {
-	let item = null;
-	owner.getHydrator().useData('media-item-id', mediaItemId.toString(), data => {
-		if (data) {
-			item = new MediaItem(data);
-		}
-	});
-
-	return item;
+export async function createMediaItemFromContentOwner(owner: ContentOwner, mediaItemId: number) {
+	let item: MediaItem | null = null;
+
+	return owner
```

Comment:
If useData is a promise, just await this.

```
return await owner.getHydrator().useData(...);
```

However, you're not really using the functionality of the async/await system, so just return without await or marking the function async. The return of the Promise will let you know it's async.

Reply:
I was having some issues making `useData(...)` work through a general function this way, so I had to revert it. That means there's slightly duplicate code, but seems to align more properly with how `owner.getHydrator().useData(...)` is intended to be used.